core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java

```diff
@@ -387,6 +376,84 @@ public void onViewUpdated(final ViewMetadata current, final ViewMetadata previou
         // NOOP
     }
 
+    /**
+     * Called when the session is up and ready. Will invoke the listeners' onSessionReady methods.
+     *
+     * @param session The session
+     */
+    @Override
+    public void onSessionReady(final Session session)
+    {
+        SchemaChangeListener.super.onSessionReady(session);
+        // prepare the NodeStateListener
+        NodeStateListener.super.onSessionReady(session);
+    }
+
+    /**
+     * Callback for when a node is added to the cluster.
+     *
+     * @param node
+     */
+    @Override
+    public void onAdd(final Node node)
+    {
+        setupConfiguration();
+    }
+
+    /**
+     * Callback for when a node is considered to be up.
+     *
+     * @param node
+     */
+    @Override
+    public void onUp(final Node node)
+    {
+        setupConfiguration();
+    }
+
+    /**
+     * Callback for when a node is considered to be down.
+     *
+     * @param node
+     */
+    @Override
+    public void onDown(final Node node)
+    {
+        setupConfiguration();
+    }
+
+    /**
+     * Callback for when a node is considered to be removed from the cluster.
+     *
+     * @param node
```

Comment:
onRemove shouldn't be needed

Reply:
When implementing NodeStateListener, onAdd/onRemove/onUp/onDown should be implemented - according to the documentation. Therefore they all just do the same thing now; e.g. reread the configuration. But, yes, onUp/onDown are the "important" ones for this case. onAdd/onRemove are more or less just called at the start/end anyways.

Reply:
If it's not needed then it's kinda pointless to have them implemented, wasting cpu etc. Just implement them and leave them empty with a comment (as we do with some other methods in this file)

Reply:
NOOP now